lib/utilities.js

```diff
@@ -329,6 +329,41 @@ const jsonParse = (buffer) => {
   }
 };
 
+const deflatObject = (source, fieldName) => {
+  const result = {};
+  for (const [key, value] of Object.entries(source)) {
+    if (typeof value === 'string') {
+      for (let i = 0; i < key.length; i++) {
+        if (inRange(key[i], 'A', 'Z')) {
```

Comment:
Do not iterate by chars, just `for (const prefix of fieldNames)` and use `startsWith`